src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -0,0 +1,1878 @@
+			framas.setParentFrame(dialog);
+			dialog.setContentPane(framas);
+			dialog.setVisible(true);
+			dialog.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+			med = framas.getSelectedMedical();
+			return med;
+		}
+		return null;
+	}
+
+	private JLabel getReferenceLabel() {
+		if (referenceLabel == null) {
+			referenceLabel = new JLabel(MessageBundle.getMessage("angal.inventory.reference.label"));
+		}
+		return referenceLabel;
+	}
+
+	private JTextField getReferenceTextField() {
+		if (referenceTextField == null) {
+			referenceTextField = new JTextField();
+			referenceTextField.setColumns(10);
+			if (inventory != null && !mode.equals("new")) {
+				referenceTextField.setText(inventory.getInventoryReference());
+			}
+		}
+		return referenceTextField;
+	}
+
+	private JLabel getStatusLabel() {
+		if (statusLabel == null) {
+			if (inventory == null) {
+				String currentStatus = InventoryStatus.draft.toString().toUpperCase();
+				statusLabel = new JLabel(currentStatus);
+				statusLabel.setForeground(Color.GRAY);
+			} else {
+				String currentStatus = inventory.getStatus().toUpperCase();
+				statusLabel = new JLabel(currentStatus);
+				if (currentStatus.equalsIgnoreCase(InventoryStatus.draft.toString())) {
+					statusLabel.setForeground(Color.GRAY);
+				}
+				if (currentStatus.equalsIgnoreCase(InventoryStatus.validated.toString())) {
+					statusLabel.setForeground(Color.BLUE);
+				}
+				if (currentStatus.equalsIgnoreCase(InventoryStatus.canceled.toString())) {
+					statusLabel.setForeground(Color.RED);
+				}
+				if (currentStatus.equalsIgnoreCase(InventoryStatus.done.toString())) {
+					statusLabel.setForeground(Color.GREEN);
+				}
+			}
+			statusLabel.setFont(new Font(statusLabel.getFont().getName(), Font.BOLD, statusLabel.getFont().getSize() + 8));
+		}
+		return statusLabel;
+	}
+
+	private JLabel getWardLabel() {
+		if (wardLabel == null) {
+			wardLabel = new JLabel(MessageBundle.getMessage("angal.inventory.selectward.label"));
+		}
+		return wardLabel;
+	}
+
+	private JComboBox<Ward> getWardComboBox() {
+		if (wardComboBox == null) {
+			wardComboBox = new JComboBox<>();
+			List<Ward> wardList;
+			try {
+				wardList = wardBrowserManager.getWards();
+			} catch (OHServiceException e) {
+				wardList = new ArrayList<>();
+				OHServiceExceptionUtil.showMessages(e);
+			}
+			if (!mode.equals("new")) {
+				String wardId = inventory.getWard();
+				for (Ward ward : wardList) {
+					if (ward.getCode().equals(wardId)) {
+						wardComboBox.addItem(ward);
+						wardSelected = ward;
+					}
+				}
+				wardComboBox.setEnabled(false);
+			} else {
+				for (Ward elem : wardList) {
+					wardComboBox.addItem(elem);
+				}
+				wardComboBox.setSelectedIndex(-1);
+			}
+
+			wardComboBox.addItemListener(itemEvent -> {
+
+				if (itemEvent.getStateChange() == ItemEvent.SELECTED) {
+					Object item = itemEvent.getItem();
+					if (item instanceof Ward) {
+						wardSelected = (Ward) item;
```

Comment:
Lines 1544 and 1545 can be replaced with:
<pre>
if (item instanceof Ward wordSelected) {
</pre>